### PR TITLE
Logging Entry new attributes

### LINKED ISSUE #1279: Add Entry#trace
#1272 adds the protobuf definitions for LogEntry trace. Add support for this value in Logging Entry.

### LINKED ISSUE #1280: Add Entry#source_location (SourceLocation)
#1272 adds the protobuf definitions for LogEntry source_location. Add support for this value in Logging Entry.

### PR DESCRIPTION
**Pull request by quartzmo**
This PR adds support for two new attributes in the protobuf definitions for LogEntry that were added in #1272.

[closes #1279, closes #1280]

### CODE REVIEW

File: google-cloud-logging/lib/google/cloud/logging/entry/source_location.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            return nil if empty?
+            Google::Logging::V2::LogEntrySourceLocation.new(
+              file:       file.to_s,
+              line:       line,
+              function:   function.to_s
+            )
+          end
+
+          ##
+          # @private New HttpRequest from a
```
Comment (blowmage) [inline google-cloud-logging/lib/google/cloud/logging/entry/source_location.rb]: `# @private New SourceLocation from a`

Comment (quartzmo) [inline google-cloud-logging/lib/google/cloud/logging/entry/source_location.rb]: I thought this comment was strange too. It's from [Operation](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-logging/lib/google/cloud/logging/entry/operation.rb#L77-L78):

```          ##
          # @private New HttpRequest from a
          # Google::Logging::V2::LogEntryOperation object.
          def self.from_grpc grpc
```

Should I just remove it in both places?

Comment (blowmage) [inline google-cloud-logging/lib/google/cloud/logging/entry/source_location.rb]: Hah! Yeah, it would be good to correct the comments in both locations. :)

Comment (blowmage) [inline google-cloud-logging/lib/google/cloud/logging/entry/source_location.rb]: Copy/paste FTW!

**Pull request merged**